### ISSUE TITLE
freenas 11.1 U2 + homebridge + ffmpeg 3.4.2

### ISSUE THREAD

**Issue by patelhiren** [KhaosT/homebridge-camera-ffmpeg#164]
Looking for some guidance on how to troubleshot video live feed not starting with the mentioned configuration.

I followed the tutorial on homebridge wiki to setup and run successfully on Freenas 11.1 U2. I can connect view the camera snapshots in the Home app. But when I start to view the live feed, the feed never starts.

I have a similar configuration working with the same camera on debian running ffmpeg 3.2. So I am guessing the ffmpeg installed in the homebridge jail has a required compile flag missing. But I am unsure which one.

Here is the snippet of the debug log. I would appreciate any guidance.
```
[3/19/2018, 8:22:05 AM] [Camera-ffmpeg] Start streaming video from Family Room Camera with 1280x720@100kBit
ffmpeg -rtsp_transport tcp -re -i rtsp://admin:admin@192.168.5.61:8001/H264/sub -map 0:0 -vcodec libx264 -pix_fmt yuv420p -r 10 -f rawvideo -preset veryfast -tune zerolatency -vf scale=1280:720 -b:v 100k -bufsize 100k -payload_type 99 -ssrc 5626954 -f rtp -srtp_out_suite AES_CM_128_HMAC_SHA1_80 -srtp_out_params 00oPLrArV1Y3p8zuKbe/KY4oRE8Nsze+HBUaw2zi srtp://172.16.1.166:58922?rtcpport=58922&localrtcpport=58922&pkt_size=1316
ffmpeg version 3.4.2 Copyright (c) 2000-2018 the FFmpeg developers
  built with FreeBSD clang version 4.0.0 (tags/RELEASE_400/final 297347) (based on LLVM 4.0.0)
  configuration: --prefix=/usr/local --mandir=/usr/local/man --datadir=/usr/local/share/ffmpeg --pkgconfigdir=/usr/local/libdata/pkgconfig --enable-shared --enable-pic --enable-gpl --enable-postproc --enable-avfilter --enable-avresample --enable-pthreads --cc=cc --disable-alsa --disable-libopencore-amrnb --disable-libopencore-amrwb --disable-libass --disable-libbs2b --disable-libcaca --disable-libcdio --disable-libcelt --disable-chromaprint --disable-libdc1394 --disable-debug --enable-htmlpages --disable-libdrm --disable-libfdk-aac --enable-ffserver --disable-libflite --enable-fontconfig --enable-libfreetype --enable-frei0r --disable-libfribidi --disable-libgme --disable-libgsm --enable-iconv --disable-libilbc --disable-jack --disable-libkvazaar --disable-ladspa --disable-libmp3lame --disable-libbluray --disable-librsvg --disable-libxml2 --enable-mmx --disable-libmodplug --disable-openal --disable-opencl --enable-libopencv --disable-opengl --disable-libopenh264 --disable-libopenjpeg --enable-optimizations --enable-libopus --disable-libpulse --enable-runtime-cpudetect --disable-librubberband --disable-sdl2 --disable-libsmbclient --disable-libsnappy --disable-sndio --disable-libsoxr --disable-libspeex --enable-sse --disable-libssh --disable-libtesseract --enable-libtheora --disable-libtwolame --enable-libv4l2 --enable-vaapi --enable-vdpau --disable-libvidstab --enable-libvorbis --disable-libvo-amrwbenc --enable-libvpx --disable-libwavpack --disable-libwebp --enable-libx264 --enable-libx265 --disable-libxcb --enable-libxvid --disable-outdev=xv --disable-libzimg --disable-libzmq --disable-libzvbi --disable-gcrypt --enable-gmp --disable-librtmp --enable-gnutls --disable-openssl --enable-version3 --disable-nonfree --disable-libmysofa

  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libavresample   3.  7.  0 /  3.  7.  0
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100
  libpostproc    54.  7.100 / 54.  7.100

Input #0, rtsp, from 'rtsp://admin:admin@192.168.5.61:8001/H264/sub':
  Metadata:
    title           : h264.mp4
    comment         : TAS-Tech Live Cast
  Duration: N/A, start: 0.060000, bitrate: N/A

    Stream #0:0: Video: h264 (Main), yuv420p(progressive), 1280x720, 16.83 tbr, 90k tbn, 180k tbc

Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> h264 (libx264))
Press [q] to stop, [?] for help

frame=    0 fps=0.0 q=0.0 size=       0kB time=-577014:32:22.77 bitrate=  -0.0kbits/s speed=N/A
[libx264 @ 0x80b417e00] VBV maxrate unspecified, assuming CBR

[libx264 @ 0x80b417e00] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 BMI2 AVX2

[libx264 @ 0x80b417e00] profile High, level 3.1

Output #0, rtp, to 'srtp://172.16.1.166:58922?rtcpport=58922&localrtcpport=58922&pkt_size=1316':
  Metadata:
    title           : h264.mp4
    comment         : TAS-Tech Live Cast
    encoder         : Lavf57.83.100
    Stream #0:0: Video: h264 (libx264), yuv420p, 1280x720, q=-1--1, 100 kb/s, 10 fps, 90k tbn, 10 tbc
    Metadata:
      encoder         : Lavc57.107.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 0/0/100000 buffer size: 100000 vbv_delay: -1

frame=   19 fps= 18 q=39.0 size=      23kB time=00:00:01.80 bitrate= 103.6kbits/s dup=0 drop=11 speed=1.68x
frame=   25 fps= 16 q=37.0 size=      29kB time=00:00:02.40 bitrate=  99.4kbits/s dup=0 drop=15 speed= 1.5x
frame=   31 fps= 15 q=36.0 size=      36kB time=00:00:03.00 bitrate=  97.7kbits/s dup=0 drop=19 speed=1.41x
frame=   36 fps= 14 q=36.0 size=      41kB time=00:00:03.50 bitrate=  94.9kbits/s dup=0 drop=23 speed=1.31x
frame=   41 fps= 13 q=37.0 size=      47kB time=00:00:04.00 bitrate=  95.7kbits/s dup=0 drop=26 speed=1.25x
frame=   47 fps= 13 q=31.0 size=      54kB time=00:00:04.60 bitrate=  96.0kbits/s dup=0 drop=30 speed=1.22x
frame=   52 fps= 12 q=29.0 size=      61kB time=00:00:05.10 bitrate=  98.2kbits/s dup=0 drop=34 speed= 1.2x
frame=   57 fps= 12 q=39.0 size=      65kB time=00:00:05.60 bitrate=  95.1kbits/s dup=0 drop=37 speed=1.17x
frame=   63 fps= 12 q=35.0 size=      71kB time=00:00:06.20 bitrate=  93.7kbits/s dup=0 drop=41 speed=1.16x
frame=   68 fps= 12 q=34.0 size=      77kB time=00:00:06.70 bitrate=  93.7kbits/s dup=0 drop=45 speed=1.14x
frame=   73 fps= 11 q=33.0 size=      82kB time=00:00:07.20 bitrate=  93.6kbits/s dup=0 drop=48 speed=1.12x
frame=   79 fps= 11 q=27.0 size=      88kB time=00:00:07.80 bitrate=  92.3kbits/s dup=0 drop=52 speed=1.12x
frame=   84 fps= 11 q=28.0 size=      92kB time=00:00:08.30 bitrate=  91.0kbits/s dup=0 drop=56 speed=1.11x
frame=   89 fps= 11 q=32.0 size=      98kB time=00:00:08.80 bitrate=  91.4kbits/s dup=0 drop=59 speed= 1.1x
frame=   95 fps= 11 q=27.0 size=     106kB time=00:00:09.40 bitrate=  92.5kbits/s dup=0 drop=63 speed= 1.1x
frame=  100 fps= 11 q=32.0 size=     115kB time=00:00:09.90 bitrate=  94.8kbits/s dup=0 drop=67 speed=1.09x
frame=  105 fps= 11 q=25.0 size=     117kB time=00:00:10.40 bitrate=  92.5kbits/s dup=0 drop=70 speed=1.08x
frame=  111 fps= 11 q=27.0 size=     124kB time=00:00:11.00 bitrate=  92.1kbits/s dup=0 drop=74 speed=1.09x
frame=  116 fps= 11 q=31.0 size=     129kB time=00:00:11.50 bitrate=  91.8kbits/s dup=0 drop=78 speed=1.08x
frame=  121 fps= 11 q=28.0 size=     135kB time=00:00:12.00 bitrate=  92.4kbits/s dup=0 drop=81 speed=1.07x
frame=  127 fps= 11 q=34.0 size=     144kB time=00:00:12.60 bitrate=  93.3kbits/s dup=0 drop=85 speed=1.07x
frame=  132 fps= 11 q=27.0 size=     146kB time=00:00:13.10 bitrate=  91.5kbits/s dup=0 drop=89 speed=1.07x
frame=  137 fps= 11 q=30.0 size=     153kB time=00:00:13.60 bitrate=  92.4kbits/s dup=0 drop=92 speed=1.06x
frame=  143 fps= 11 q=25.0 size=     160kB time=00:00:14.20 bitrate=  92.1kbits/s dup=0 drop=96 speed=1.06x
frame=  148 fps= 11 q=27.0 size=     164kB time=00:00:14.70 bitrate=  91.6kbits/s dup=0 drop=100 speed=1.06x
frame=  153 fps= 11 q=29.0 size=     170kB time=00:00:15.20 bitrate=  91.7kbits/s dup=0 drop=103 speed=1.06x
frame=  159 fps= 11 q=21.0 size=     180kB time=00:00:15.80 bitrate=  93.5kbits/s dup=0 drop=107 speed=1.06x
frame=  164 fps= 11 q=32.0 size=     185kB time=00:00:16.30 bitrate=  92.8kbits/s dup=0 drop=111 speed=1.05x
frame=  169 fps= 11 q=26.0 size=     188kB time=00:00:16.80 bitrate=  91.7kbits/s dup=0 drop=114 speed=1.05x
frame=  175 fps= 11 q=28.0 size=     197kB time=00:00:17.40 bitrate=  92.9kbits/s dup=0 drop=118 speed=1.05x
frame=  180 fps= 11 q=28.0 size=     203kB time=00:00:17.90 bitrate=  92.7kbits/s dup=0 drop=122 speed=1.05x
frame=  185 fps= 11 q=27.0 size=     206kB time=00:00:18.40 bitrate=  91.8kbits/s dup=0 drop=125 speed=1.05x
frame=  191 fps= 11 q=23.0 size=     212kB time=00:00:19.00 bitrate=  91.6kbits/s dup=0 drop=129 speed=1.05x
frame=  196 fps= 11 q=27.0 size=     219kB time=00:00:19.50 bitrate=  91.8kbits/s dup=0 drop=133 speed=1.04x
frame=  201 fps= 10 q=27.0 size=     226kB time=00:00:20.00 bitrate=  92.5kbits/s dup=0 drop=136 speed=1.04x
frame=  207 fps= 10 q=23.0 size=     231kB time=00:00:20.60 bitrate=  91.8kbits/s dup=0 drop=140 speed=1.04x
frame=  212 fps= 10 q=26.0 size=     237kB time=00:00:21.10 bitrate=  91.8kbits/s dup=0 drop=144 speed=1.04x
frame=  217 fps= 10 q=26.0 size=     242kB time=00:00:21.60 bitrate=  91.8kbits/s dup=0 drop=147 speed=1.04x
frame=  223 fps= 10 q=20.0 size=     254kB time=00:00:22.20 bitrate=  93.7kbits/s dup=0 drop=151 speed=1.04x
frame=  228 fps= 10 q=26.0 size=     257kB time=00:00:22.70 bitrate=  92.7kbits/s dup=0 drop=155 speed=1.04x
frame=  233 fps= 10 q=29.0 size=     262kB time=00:00:23.20 bitrate=  92.6kbits/s dup=0 drop=158 speed=1.04x
frame=  239 fps= 10 q=24.0 size=     268kB time=00:00:23.80 bitrate=  92.2kbits/s dup=0 drop=162 speed=1.04x
frame=  244 fps= 10 q=26.0 size=     275kB time=00:00:24.30 bitrate=  92.6kbits/s dup=0 drop=166 speed=1.04x
frame=  249 fps= 10 q=25.0 size=     284kB time=00:00:24.80 bitrate=  93.8kbits/s dup=0 drop=169 speed=1.03x
[libx264 @ 0x80b417e00] VBV underflow (frame 251, -9880 bits)

frame=  255 fps= 10 q=45.0 size=     296kB time=00:00:25.40 bitrate=  95.3kbits/s dup=0 drop=173 speed=1.04x
frame=  260 fps= 10 q=38.0 size=     302kB time=00:00:25.90 bitrate=  95.6kbits/s dup=0 drop=177 speed=1.03x
frame=  265 fps= 10 q=36.0 size=     308kB time=00:00:26.40 bitrate=  95.6kbits/s dup=0 drop=180 speed=1.03x
frame=  271 fps= 10 q=36.0 size=     314kB time=00:00:27.00 bitrate=  95.2kbits/s dup=0 drop=184 speed=1.03x
frame=  276 fps= 10 q=35.0 size=     320kB time=00:00:27.50 bitrate=  95.3kbits/s dup=0 drop=188 speed=1.03x
frame=  281 fps= 10 q=34.0 size=     326kB time=00:00:28.00 bitrate=  95.3kbits/s dup=0 drop=191 speed=1.03x
frame=  287 fps= 10 q=37.0 size=     332kB time=00:00:28.60 bitrate=  95.0kbits/s dup=0 drop=195 speed=1.03x
frame=  292 fps= 10 q=34.0 size=     335kB time=00:00:29.10 bitrate=  94.2kbits/s dup=0 drop=199 speed=1.03x
frame=  297 fps= 10 q=33.0 size=     339kB time=00:00:29.60 bitrate=  93.9kbits/s dup=0 drop=202 speed=1.03x
frame=  298 fps= 10 q=31.0 Lsize=     341kB time=00:00:29.70 bitrate=  94.1kbits/s dup=0 drop=204 speed=1.02x
video:339kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.663811%

[libx264 @ 0x80b417e00] frame I:2     Avg QP:43.18  size:  7016
[libx264 @ 0x80b417e00] frame P:296   Avg QP:27.91  size:  1125
[libx264 @ 0x80b417e00] mb I  I16..4: 47.1% 52.8%  0.1%

[libx264 @ 0x80b417e00] mb P  I16..4:  0.6%  0.4%  0.0%  P16..4:  4.9%  0.6%  0.1%  0.0%  0.0%    skip:93.4%
[libx264 @ 0x80b417e00] 8x8 transform intra:44.7% inter:16.8%
[libx264 @ 0x80b417e00] coded y,uvDC,uvAC intra: 18.0% 26.8% 0.8% inter: 1.0% 1.6% 0.0%
[libx264 @ 0x80b417e00] i16 v,h,dc,p: 34% 33% 29%  5%
[libx264 @ 0x80b417e00] i8 v,h,dc,ddl,ddr,vr,hd,vl,hu: 14% 17% 63%  1%  1%  1%  1%  1%  1%
[libx264 @ 0x80b417e00] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 26% 30% 27%  2%  2%  3%  5%  4%  2%
[libx264 @ 0x80b417e00] i8c dc,h,v,p: 82%  9%  8%  1%
[libx264 @ 0x80b417e00] Weighted P-Frames: Y:0.0% UV:0.0%
[libx264 @ 0x80b417e00] kb/s:93.14

Exiting normally, received signal 15.

[3/19/2018, 8:22:36 AM] [Camera-ffmpeg] Stopped streaming

```

**Comment by patelhiren** [comment]
I have an update regarding this problem I was running into. It turns out the "-tune zerolatency" parameter being passed to ffmpeg was causing some problems. I am unsure what the problem with this parameter is, but removing that parameter made the live video work.

However I found that the performace of live video was abysmal and would only update the frame once every 15 to 20 seconds. What I found if I pass 

```-preset ultrafast -profile:v main -level 1```

to ffmpeg the live feed works well. So I have a version where I can set **additionalCommandline** as a config setting and pass those values.

I also found that if I did not setup a stderr listener on ffmpeg process, my stream would work for about a minute to minute and a half and then stop responding. So I reworked the code to always setup the listner and only log to console if debug config is set.

With these changes everything seems to work fine.
